server/stream.go

```diff
@@ -2029,6 +2030,39 @@ func (mset *stream) updateWithAdvisory(config *StreamConfig, sendAdvisory bool)
 	return nil
 }
 
+// Sets the configuration for this stream.
+// This is called under mset.mu.Lock(), but this function will also acquire
+// mset.cfgMu lock. This is so that in places where mset.mu cannot be acquired
+// (like many cases in consumer.go where code is under the consumer's lock),
+// but the stream's configuration needs to be inspected, the caller can call
+// mset.getCfg() to get (safely) a copy of the stream's configuration.
+func (mset *stream) setCfg(cfg *StreamConfig) {
+	mset.cfgMu.Lock()
+	mset.cfg = *cfg
+	mset.cfgMu.Unlock()
+}
+
+// The stream configuration is a structure. Normally, code that would do
+// something like cfg := mset.cfg would get a copy of that structure. However,
+// this would race with code that assigns a new congiguration to the stream
+// (think stream update). The update will be done using setCfg(), and therefore,
+// to avoid races, callers that need the stream's congiguration should call
+// this function, or protect using mset.cfgMu read lock.
+func (mset *stream) getCfg() StreamConfig {
```

Comment:
If it is agreed that call to `getCfg()` in `addConsumerWithAssignment()` is not needed, then this method may not be needed at all (for now at least).

Reply:
I have removed it, and setCfg() too (I did inline it since there was only one place we were using it).

Reply:
I don't think it is needed. With it not there can you see the race?